### PR TITLE
feat: Sushiswap

### PR DESCRIPTION
**Pull request by dekz**
## Description

TODO deploy SushiSwap bridge

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/asset-proxy/contracts/src/bridges/SushiSwapBridge.sol [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    )
+        external
+        returns (bytes4 success)
+    {
+        // hold variables to get around stack depth limitations
+        TransferState memory state;
+
+        // Decode the bridge data to get the `fromTokenAddress`.
+        // solhint-disable indent
+        (state.path, state.router) = abi.decode(bridgeData, (address[], address));
```
Comment (dekz) [inline contracts/asset-proxy/contracts/src/bridges/SushiSwapBridge.sol:69]: encoded the router incase anything changes in the short term. Might remove later idk

File: packages/asset-swapper/contracts/src/SushiSwapSampler.sol [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    uint256 constant private SUSHISWAP_CALL_GAS = 150e3; // 150k
+
+    /// @dev Sample sell quotes from SushiSwap.
+    /// @param router Router to look up tokens and amounts
+    /// @param path Token route. Should be takerToken -> makerToken
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromSushiSwap(
+        address router,
```
Comment (dekz) [inline packages/asset-swapper/contracts/src/SushiSwapSampler.sol:39]: encoded the router incase anything changes in the short term. Might remove later idk

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts [hunk 3]
```diff
@@ -185,6 +187,8 @@ export const MAINNET_KYBER_TOKEN_RESERVE_IDS: { [token: string]: string } = {
         '0xaa42414e44000000000000000000000000000000000000000000000000000000',
 };
 
+export const MAINNET_SUSHI_SWAP_ROUTER = '0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F';
```
Comment (dekz) [inline packages/asset-swapper/src/utils/market_operation_utils/constants.ts:190]: this is the router. Might remove later idk

File: packages/asset-swapper/src/utils/market_operation_utils/orders.ts [hunk 4]
```diff
@@ -223,6 +224,9 @@ function getBridgeAddressFromFill(fill: CollapsedFill, opts: CreateOrderFromPath
             return opts.contractAddresses.uniswapBridge;
         case ERC20BridgeSource.UniswapV2:
             return opts.contractAddresses.uniswapV2Bridge;
+        case ERC20BridgeSource.SushiSwap:
+            return '0xaaaaaa1111111111111111111111111111111111';
```
Comment (dekz) [inline packages/asset-swapper/src/utils/market_operation_utils/orders.ts]: Replace with deployed bridge

**Review by dorothy-zbornak — APPROVED** [review]
ye I like it like that
tenor

**Pull request merged**